### DISCUSSION TITLE
Handling CASE, DECRYPT and UNION queries in PRISMA

### DISCUSSION THREAD

**Discussion by proxima-b-alpha** [prisma/prisma#3347] [Q&A] [answered]
Hi, I have few questions for which i was not able to find answers in PRISMA docs.

DB: MySQL

How do we handle these query scenaries in PRISMA?

CASE Scenario:

```sql
SELECT brands.id, brands.brand, brands.established_date,
CASE items.active WHEN true THEN items.quantity ELSE 0 END as itemQuantities
FROM tshirts.brands
LEFT JOIN tshirts.items items ON items.id = brands.id
WHERE brands.id = '1';

```
DECRYPT Scenario:

```sql
SELECT CAST(AES_DECRYPT(brands.brand, <key>) as char) as brand, brands.established_date
FROM tshirts.brands
WHERE brands.id = '1';
```

UNION Scenario:

```sql
SELECT City FROM Customers
UNION
SELECT City FROM Suppliers
ORDER BY City;
```

**Comment by ryands17** [comment, marked as answer]
Hey @proxima-b-alpha 👋 
Unfortunately all these queries need to be handled via `prisma.queryRaw` itself as Prisma currently doesn't support native query operators like `CASE` and `DECRYPT`.

Although you could create a `UNION` by fetching from the two tables and merging them programatically in your code.

It would also be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with your use case.

**Reply by janpio** [reply]
Best 1 feature request per type of query - then it is easiest to categorize those for us and push them forward.

**Reply by proxima-b-alpha** [reply]
Thank You @ryands17

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
